src/database.ts

```diff
@@ -919,34 +1097,32 @@ class Database extends ServiceObject {
     optionsOrCallback?: TimestampBounds | GetSnapshotCallback,
     cb?: GetSnapshotCallback
   ): void | Promise<[Snapshot]> {
-    let options: TimestampBounds;
-    let callback: GetSnapshotCallback;
-
-    if (is.fn(optionsOrCallback)) {
-      callback = optionsOrCallback! as GetSnapshotCallback;
-      options = {};
-    } else {
-      options = optionsOrCallback as TimestampBounds;
-      callback = cb!;
-    }
+    const callback =
+      typeof optionsOrCallback === 'function'
+        ? (optionsOrCallback as GetSnapshotCallback)
+        : cb!;
```

Comment:
```suggestion
        : cb;
```
to accommodate for Promise version